### DISCUSSION TITLE
Does GCNConv take duplicate edges into account?

### DISCUSSION THREAD

**Discussion by ZRH0308** [pyg-team/pytorch_geometric#4333] [Q&A] [answered]
Hi,

I am new to geometrics, I need your help with confusion that may be stupid but important to me.
 
I was wondering when I give GCN.forward() an input like below,

```
edge_index = torch.tensor([[0, 1, 1, 1, 1, 0],
                           [1, 2, 2, 2, 0, 1]], dtype=torch.long)

x = torch.tensor([[-1], [0], [1]], dtype=torch.float)
```

How will GCNConv( x, edge_index)  process the duplicate edges in edge_index? Merge them or just drop the duplicates?
In other words, in its view, my input will be treated as which of the following matrices? ( I don't add self-loop here)

```
tensor([[[0., 2., 0.],
         [1., 0., 3.],
         [0., 0., 0.]]])
```

OR

```
tensor([[[0., 1., 0.],
         [1., 0., 1.],
         [0., 0., 0.]]])
```

？

Thank you very much !

**Comment by rusty1s** [comment, marked as answer]
We will not remove duplicate edges, but instead interpret them as two edges. As such, it will be treated as
```
tensor([[[0., 2., 0.],
         [1., 0., 3.],
         [0., 0., 0.]]])
```

**Reply by ZRH0308** [reply]
Thanks for your reply !!
